crates/octez/src/async/protocol.rs

```diff
@@ -216,6 +233,23 @@ impl ProtocolParameterBuilder {
         );
         Ok(())
     }
+
+    fn merge_bootstrap_smart_rollups(
+        &mut self,
+        json: &mut serde_json::Map<String, Value>,
+    ) -> anyhow::Result<()> {
+        let mut rollups = BootstrapSmartRollups::default();
+        if let Some(value) = json.get("bootstrap_smart_rollups") {
+            let existing_rollups = serde_json::from_value(value.clone())?;
+            rollups.merge(&existing_rollups);
+        }
+        rollups.merge(&self.bootstrap_smart_rollups);
+        json.insert(
+            "bootstrap_smart_rollups".to_owned(),
+            serde_json::to_value(rollups)?,
+        );
```

Comment:
Might be nice to generalize the merge of bootstrap containers in another PR. I think they just need to implement a `trait Mergeable<T> { fn merge(&self) -> Result<T> }` and this fn could be generalized to:
```
fn merge<T: Mergeable + Serialize + Default>(value: T, json: &mut serde_json::Map<String, Value>, key: &str) -> Result<()> 
```